### DISCUSSION TITLE
When using both `defaultVaules` and `values`, useForm `getValues` returns defaultsValues in the first place.

### DISCUSSION THREAD

**Discussion by t1m0n** [react-hook-form/react-hook-form#11397] [Q&A]
Hello! 
I have a typical situation when I fetch some data and pass this data to the `useForm`.  I also have `defaultValues` set along with `values`. The problem is that when my request is loaded I show the page with a form and it displays `defaultValues` for a moment and replaces them with actual data after.

Even if using sync data, `useForm` anyway for a moment uses `defaultValues` instead of `values`

**Example**
```jsx
const Form = () => {
  const form = useForm({
    defaultValues: {
      name: 'name default',
    },
    values: {
      name: 'name current',
    },
  });

  console.log(form.getValues());
};
```
Live example - https://stackblitz.com/edit/stackblitz-starters-lkabsa?file=src%2FApp.tsx

The code above firstly logs `name default` and after `name current`.

So the question is, is where anyway to get `values` in the first place if both `defaultValues` and `values` are set?

Thank you!

**Comment by chanthinh** [comment]
Hi @t1m0n,
I think it can not get `values` in the first place if both `defaultValues` and `values` are set.
Because, `It will return defaultValues from useForm before the initial render`.
DOC: https://www.react-hook-form.com/api/useform/getvalues/
